internal/plugin/plugin.go

```diff
@@ -93,12 +95,39 @@ func NewPluginClient(connector string, version string, logger hclog.Logger) (Plu
 		return nil, fmt.Errorf("the plugin (%s) doesn't correctly implement the necessary interfaces", connector)
 	}
 
+	currentVersion := version.CurrentCliVersionInformation{}
+	if _, err = isPluginCompatibleWithCli(&currentVersion, is, logger); err != nil {
+		logger.Error("Plugin not compatible: %s", err.Error())
+		return nil, err
+	}
+
 	info := is.PluginInfo()
 	logger.Debug("Using plugin: " + info.String())
 
 	return pci, nil
 }
 
+func isPluginCompatibleWithCli(currentCliInformation version.CliVersionInformation, pluginInfo plugin2.Info, logger hclog.Logger) (bool, error) {
+	if currentCliInformation.GetCliVersion().Equal(version.DevVersion) {
```

Comment:
Maybe we should not skip this check for dev version but rather consider it as the latest cli version. But would be good to find incompatible plugins asap?